pkg/agent/core/ngt/handler/grpc/handler_test.go

```diff
@@ -1176,84 +1188,1309 @@ func Test_server_Insert(t *testing.T) {
+
+			- case 6: Insert nil insert request fail
+				* IncompatibleDimensionSize error will be returned.
+			- case 7: Insert nil vector fail
+				* IncompatibleDimensionSize error will be returned.
+			- case 8: Insert empty insert vector fail
+				* IncompatibleDimensionSize error will be returned.
+
+		- Decision Table Testing
+			- duplicated ID, duplicated vector, duplicated ID & vector
+				- case 1.1: Insert duplicated request fail when SkipStrictExistCheck is false (duplicated ID)
+					* AlreadyExists error will be returned.
+				- case 1.2: Insert duplicated request success when SkipStrictExistCheck is false (duplicated vector)
+				- case 1.3: Insert duplicated request fail when SkipStrictExistCheck is false (duplicated ID & vector)
+				- case 2.1: Insert duplicated request fail when SkipStrictExistCheck is true (duplicated ID)
+					* SkipStrictExistCheck flag is not used in agent handler, so the result is same as case 1.
+				- case 2.2: Insert duplicated request success when SkipStrictExistCheck is true (duplicated vector)
+				- case 2.3: Insert duplicated request fail when SkipStrictExistCheck is true (duplicated ID & vector)
+	*/
+	tests := []test{
+		// Equivalence Class Testing
+		func() test {
+			return test{
+				name: "Equivalence Class Testing case 1.1: Insert vector success (vector type is uint8)",
+				args: args{
+					ctx: ctx,
+					req: &payload.Insert_Request{
+						Vector: &payload.Object_Vector{
+							Id:     id,
+							Vector: intVec,
+						},
+					},
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    intVecDim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Uint8.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				want: want{
+					wantRes: &payload.Object_Location{
+						Name: name,
+						Uuid: id,
+						Ips:  []string{ip},
+					},
+				},
+			}
+		}(),
+		func() test {
+			return test{
+				name: "Equivalence Class Testing case 1.2: Insert vector success (vector type is float32)",
+				args: args{
+					ctx: ctx,
+					req: &payload.Insert_Request{
+						Vector: &payload.Object_Vector{
+							Id:     id,
+							Vector: f32Vec,
+						},
+					},
+				},
+				fields: fields{
+					name: name,
+					ip:   ip,
+					svcCfg: &config.NGT{
+						Dimension:    f32VecDim,
+						DistanceType: ngt.Angle.String(),
+						ObjectType:   ngt.Float.String(),
+						KVSDB:        &config.KVSDB{},
+						VQueue:       &config.VQueue{},
+					},
+					svcOpts: []service.Option{
+						service.WithEnableInMemoryMode(true),
+					},
+				},
+				want: want{
+					wantRes: &payload.Object_Location{
+						Name: name,
+						Uuid: id,
+						Ips:  []string{ip},
+					},
+				},
+			}
+		}(),
+		func() test {
+			vec := []float32{1, 2, 3, 4, 5, 6, 7}
```

Comment:
can you use generator instead of it?

Reply:
I have added few success cases to use the generator in `Equivalence Class Testing case 3.1 - 4.2`, do you mean I should use the generator in this fail case also?